clippy_lints/src/manual_is_ascii_check.rs

```diff
@@ -0,0 +1,142 @@
+use rustc_ast::LitKind::{Byte, Char};
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, PatKind, RangeEnd};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_semver::RustcVersion;
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::{def_id::DefId, sym};
+
+use clippy_utils::{diagnostics::span_lint_and_sugg, macros::root_macro_call, meets_msrv, msrvs, source::snippet};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Suggests to use dedicated built-in methods,
+    /// `is_ascii_(lowercase|uppercase|digit)` for checking on corresponding ascii range
+    ///
+    /// ### Why is this bad?
+    /// Readability.
```

Comment:
I would expand this a bit further, like:

```suggestion
    /// Using the build in functions is more readable and makes it
    /// clear that it's not a specific subset of characters, but all
    /// ASCII (lowercase|uppercase|digit) characters.
```

Reply:
Thank you, it looks so nice!
I'm not native English speaker so not very confident though, it would be better with `Using built-in functions is ...`?